packages/flame/test/game/mixins/has_parent_test.dart

```diff
@@ -0,0 +1,32 @@
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame_test/flame_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+class ParentComponent extends Component {}
+
+class DifferentComponent extends Component {}
+
+class TestComponent extends Component with HasParent<ParentComponent> {}
+
+void main() {
+  group('HasParent', () {
+    testWithFlameGame('successfully sets the parent link', (game) async {
+      final parent = ParentComponent();
+      final component = TestComponent();
+
+      await parent.add(component);
+      await game.add(parent);
+
+      expect(component.parent, isA<ParentComponent>());
```

Comment:
This would have succeeded even if there was no `HasParent` mixin.
Perhaps we could add some `ParentComponent { int foo; }` property, and then check that `component.parent.foo` can be read without an explicit type-cast?

Reply:
The problem with that is, we can't. It would fail before that, and if it didn't fail it will be correct anyway. It wouldn't test anything really. This whole test doesn't really test anything except that there was no error from the `add`